libraries/config/mechanical_material_data.hpp

```diff
@@ -35,26 +35,97 @@ struct mechanical_material_data<T, 1> final {
+            }
+        } else
+            throw std::domain_error{"The " + name + " should be either a number in the isotropic case, "
+                                    "or an array of size 2 in the anisotropic"};
+    }
+    void calculate_elasticity_parameters(const std::array<size_t, 2>& null_E, const std::array<size_t, 2>& null_nu) {
+        // youngs_modulus[1] * poissons_ratio[0] = Ey * nuxy = Ex * nuyx = youngs_modulus[0] * poissons_ratio[1]
+        if (null_E[0] == 0 && null_E[1] == 1) {
+            youngs_modulus[1] = youngs_modulus[0] * poissons_ratio[1] / poissons_ratio[0];
+            return;
+        }
+        if (null_E[0] == 1 && null_E[1] == 0) {
+            youngs_modulus[0] = youngs_modulus[1] * poissons_ratio[0] / poissons_ratio[1];
+            return;
+        }
+        if (null_nu[0] == 0 && null_nu[1] == 1) {
+            poissons_ratio[1] = youngs_modulus[1] * poissons_ratio[0] / youngs_modulus[0];
+            return;
+        }
+        if (null_nu[0] == 1 && null_nu[1] == 0) {
+            poissons_ratio[0] = youngs_modulus[0] * poissons_ratio[1] / youngs_modulus[1];
+            return;
+        }
+    }  
+    bool check_elasticity_parameters(const std::array<size_t, 2>& null_E, const std::array<size_t, 2>& null_nu) {
+        const size_t null = null_nu[0] + null_nu[1] + null_E[0] + null_E[1];
+        if (null == 1) return false;
+        return true;
+    }
+    bool elasticity_parameters_correctness() {
+        if (poissons_ratio[0] <= 0 || poissons_ratio[0] >= 0.5)
+            return true;
+        if (poissons_ratio[1] <= 0 || poissons_ratio[1] >= 0.5)
+            return true;
+        if (youngs_modulus[0] <= 0 || youngs_modulus[1] <= 0)
+            return true;
+        if (shear_modulus <= 0)
+            return true;
+        return false;
+    }
+public:
     static constexpr std::string_view Prefix = "mechanical";
-
-    T youngs_modulus = T{1}; // required
-    T poissons_ratio = T{0.3}; // required
+    material_t material = material_t::ISOTROPIC;        // not json field
+    std::array<T, 2> youngs_modulus = {T{1},   T{1}};   // required
+    std::array<T, 2> poissons_ratio = {T{0.3}, T{0.3}}; // required
+    T shear_modulus = T{1};
     T thermal_expansion = T{0}; // optional
 
     explicit constexpr mechanical_material_data() noexcept = default;
     explicit mechanical_material_data(const nlohmann::json& config, const std::string& path = {}) {
         const std::string right_part = append_access_sign(path);
-        check_required_fields(config, { "youngs_modulus", "poissons_ratio" }, right_part);
-        youngs_modulus = config["youngs_modulus"].get<T>();
-        poissons_ratio = config["poissons_ratio"].get<T>();
+        check_required_fields(config, { "youngs_modulus", "poissons_ratio"}, right_part);
+        check_optional_fields(config, { "shear_modulus" }, right_part); 
+        std::array<size_t, 2> null_E {0, 0}, null_nu {0, 0};
```

Comment:
```suggestion
        std::array<bool, 2> null_E = {false, false};
        std::array<bool, 2> null_nu = {false, false};
```
size_t здесь выглядит избыточно, для флажков обычно используют bool. В целом, здесь можно заморочиться и вместо std::array использовать std::bitset, там же есть удобные методы по типу .all(), .any() и т.д.